lib/queuePopulator/KafkaLogConsumer/LogConsumer.js

```diff
@@ -0,0 +1,166 @@
+const kafka = require('node-rdkafka');
+const async = require('async');
+const ListRecordStream = require('./ListRecordStream');
+
+// maximum time to wait for consumer group to rebalance (ms)
+const maxDelayRebalance = 120000;
+
+class LogConsumer {
+
+    /**
+     * @constructor
+     * @param {Object} kafkaConfig queue populator kafka config
+     * @param {string} kafkaConfig.hosts kafka hosts
+     * @param {string} kafkaConfig.topic kafka oplog topic
+     * @param {string} kafkaConfig.groupId consumer group id
+     * @param {Logger} logger logger
+     */
+    constructor(kafkaConfig, logger) {
+        const { hosts, topic, groupId } = kafkaConfig;
+        this._kafkaHosts = hosts;
+        this._topic = topic;
+        this._groupId = groupId;
+        this._log = logger;
+        this._offsets = {};
+    }
+
+    /**
+     * Get partition offsets
+     * @returns {string} stored partition offsets
+     */
+    _getOffset() {
+        // Format: [{ topic: 'oplog-topic', partition: 0, offset: 0 }]
+        return JSON.stringify(this._offsets);
+    }
+
+    /**
+     * Connects consumer to kafka and subscribes
+     * to oplog topic
+     * @param {Function} done callback
+     * @returns {undefined}
+     */
+    setup(done) {
+        // partition offsets will be managed by kafka
+        const consumerParams = {
+            'enable.auto.offset.store': true,
+            'metadata.broker.list': this._kafkaHosts,
+            'group.id': this._groupId,
+        };
+        const topicParams = {
+            'auto.offset.reset': 'earliest',
+            'enable.auto.commit': true,
+        };
+        this._consumer = new kafka.KafkaConsumer(consumerParams, topicParams);
+        this._consumer.connect();
+        this._consumer.on('ready', () => {
+            this._consumer.subscribe([this._topic]);
+            return done();
+        });
+    }
+
+    /**
+     * Waits for consumer group to rebalance
+     * @param {number} wait amount of time to wait
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    _waitForAssignment(wait, cb) {
+        setTimeout(() => {
+            /**
+             * assignements contain the partitions
+             * assigned to this consumer, they are only
+             * set once consumer group is balanced
+             */
```

Comment:
Should not be a "doc" comment?

```suggestion
             // assignements contain the partitions
             // assigned to this consumer, they are only
             // set once consumer group is balanced
```